apps/rehash.c

```diff
@@ -168,6 +168,11 @@ static int add_entry(enum Type type, unsigned int hash, const char *filename,
         *ep = nilhentry;
         ep->old_id = ~0;
         ep->filename = OPENSSL_strdup(filename);
+        if (ep->filename == NULL) {
+            OPENSSL_free(ep);
```

Comment:
Perhaps some BIO_printf with an error message would be a good idea?

Reply:
OK, I have amended commit message and submitted a fixup.